### PR TITLE
using system suitesparse package

### PR DESCRIPTION
**Pull request by huberya**
@HannesSommer 
Is this the idea how to use it?
(I would need to add the same for aslam_backend_expressions, as this fails with only system suitesparse)

### CODE REVIEW

**Comment by huberya** [comment]
this should be working aswell (once suitesparse is merged)

**Comment by HannesSommer** [comment]
Changes look good. What is missing is the ci/prepare-build.sh I had in this branch:  https://github.com/ethz-asl/aslam_optimizer/tree/test/use_suitesparse_system_lib . Could add it and make Jenkins happy?

**Comment by HannesSommer** [comment]
Wait, actually I'm confuse that you have to put it to all libraries instead of only for` sparse_block_matrix` why is that?

**Comment by HannesSommer** [comment]
Ah, because of the transitive include folder dependencies. What about reexporting the headers from suitesparse from sparse_block_matrix with something like adding `suitesparse_INCLUDE_DIRS` to its `cs_export`?

**Comment by huberya** [comment]
I tried your suggestion, I hope I got that right

**Comment by HannesSommer** [comment]
Yes, that is exactly what I had in mind. Thanks. It also seems to work :).

File: sparse_block_matrix/CMakeLists.txt [hunk 1]
```diff
@@ -7,6 +7,7 @@ catkin_simple(ALL_DEPS_REQUIRED)
 # enable warnings
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wno-deprecated-register") # deprecated-register warnings are cause by old Eigen versions
 
+
```
Comment (HannesSommer) [inline sparse_block_matrix/CMakeLists.txt:10]:
```suggestion
```

**Pull request merged**